core/src/mindustry/mod/Scripts.java

```diff
@@ -116,6 +118,47 @@ public Music loadMusic(String soundName){
         return sound;
     }
 
+    /** Ask the user to select a file to read for a certain purpose like "Please upload a sprite"
+        Extension can be null to select any file. */
+    public void readFile(String purpose, String ext, Cons<String> string){
+        selectFile(true, purpose, ext, fi -> cons.get(fi.readString()));
+    }
+
+    /** readFile but for a byte[] */
+    public void readBinFile(String purpose, String ext, Cons<byte[]> cons){
+        selectFile(true, purpose, ext, fi -> cons.get(fi.readBytes()));
+    }
+
+    /** Ask the user to write a file.
+        Extension can be null to allow any filename. */
+    public void writeFile(String purpose, String ext, String contents){
+        if(contents == null) contents = "";
+        selectFile(false, purpose, ext, fi -> fi.writeString(contents));
+    }
+
+    /** writeFile but for a byte[] */
+    public void writeBinFile(String purpose, String ext, byte[] contents){
+        if(contents == null) contents = new byte[0];
+        selectFile(false, purpose, ext, fi -> fi.writeBytes(contents));
+    }
+
+    private void selectFile(boolean open, String purpose, String ext, Cons<Fi> cons){
+        purpose = purpose.startsWith("@") ? Core.bundle.get(purpose.substring(1)) : purpose;
```

Comment:
actually, Arc also uses $ for bundle shortcurs.
```suggestion
        purpose = purpose.startsWith("@") ||  purpose.startsWith("$") ? Core.bundle.get(purpose.substring(1)) : purpose;
```

Reply:
anuke changed that, see Platform using @save/@open

Reply:
I said Arc *also* uses $, see [Label#104](https://github.com/Anuken/arc/blob/master/arc-core/src/arc/scene/ui/Label.java#L104).